japicmp/src/main/java/japicmp/compat/CompatibilityChanges.java

```diff
@@ -584,7 +584,25 @@ public Integer callback(JApiClass superclass, Map<String, JApiClass> classMap, J
 					} else if (superClassChangedFromObject) {
 						addCompatibilityChange(superclass, JApiCompatibilityChange.SUPERCLASS_ADDED);
 					} else {
-						addCompatibilityChange(superclass, JApiCompatibilityChange.SUPERCLASS_CHANGED);
+						// check if the old superclass is still an ancestor of the new superclass
+						List<JApiSuperclass> ancestors = new ArrayList<>();
+						final List<JApiSuperclass> matchingAncestors = new ArrayList<>();
+						forAllSuperclasses(jApiClass, classMap, ancestors, new OnSuperclassCallback<JApiSuperclass>() {
+							@Override
+							public JApiSuperclass callback(JApiClass clazz, Map<String, JApiClass> classMap, JApiChangeStatus changeStatusOfSuperclass) {
+								JApiSuperclass ancestor = clazz.getSuperclass();
+								if (ancestor.getNewSuperclassName().isPresent() && ancestor.getNewSuperclassName().get().equals(superclass.getOldSuperclassName().get())) {
```

Comment:
If that was the case, then we we would have duplicates in the class hierarchy for the new version of the code being compared. While that's not a reason for a ClassCircularityError to be thrown (as per Section 13.4.4. Superclasses and Superinterfaces), it would at least make it possible. Maybe we should add a check to detect that in another PR?

In this case, we care about the old superclass, since that's the one that's been removed from the jApiClass we're checking, and we're making sure it's still in the inheritance heirarchy. If it's not, then the change is not binary compatible, since jApiClass must continue to be assignable to oldSuperclass.